inc/classes/class-logger.php

```diff
@@ -0,0 +1,479 @@
+			return Registry::$logger_name();
+		}
+
+		/**
+		 * File handler.
+		 * HTML formatter is used.
+		 *
+		 * Note, if Inline formatter is used:
+		 *     - Keep default formats, allow line breaks in messages, hide empty contexts and extras.
+		 *     - `$formatter = new LineFormatter( null, null, true, true );`.
+		 */
+		$handler   = new StreamHandler( static::get_log_file_path(), $log_level );
+		$formatter = new HtmlFormatter();
+
+		$handler->setFormatter( $formatter );
+
+		/**
+		 * Thanks to the processors, add data to each log:
+		 * - `debug_backtrace()` (exclude this class).
+		 */
+		$trace_processor = new Processor\IntrospectionProcessor( $log_level, [ get_called_class() ] );
+
+		// Create the logger.
+		$logger = new Monologger( $logger_name, [ $handler ], [ $trace_processor ] );
+
+		// Store the logger.
+		Registry::addLogger( $logger );
+
+		return $logger;
+	}
+
+
+	/** ----------------------------------------------------------------------------------------- */
+	/** LOG FILE ================================================================================ */
+	/** ----------------------------------------------------------------------------------------- */
+
+	/**
+	 * Get the path to the log file.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return string
+	 */
+	public static function get_log_file_path() {
+		return WP_CONTENT_DIR . '/wp-rocket-config/' . static::LOG_FILE_NAME;
+	}
+
+	/**
+	 * Get the log file contents.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return string|object The file contents on success. A WP_Error object on failure.
+	 */
+	public static function get_log_file_contents() {
+		$filesystem = \rocket_direct_filesystem();
+		$file_path  = static::get_log_file_path();
+
+		if ( ! $filesystem->exists( $file_path ) ) {
+			return new \WP_Error( 'no_file', __( 'The log file does not exist.', 'rocket' ) );
+		}
+
+		$contents = $filesystem->get_contents( $file_path );
+
+		if ( false === $contents ) {
+			return new \WP_Error( 'file_not_read', __( 'The log file could not be read.', 'rocket' ) );
+		}
+
+		return $contents;
+	}
+
+	/**
+	 * Get the log file size and number of entries.
+	 *
+	 * @since  3.1.4
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return array|object An array of statistics on success. A WP_Error object on failure.
+	 */
+	public static function get_log_file_stats() {
+		$filesystem = \rocket_direct_filesystem();
+		$file_path  = static::get_log_file_path();
+
+		if ( ! $filesystem->exists( $file_path ) ) {
+			return new \WP_Error( 'no_file', __( 'The log file does not exist.', 'rocket' ) );
+		}
+
+		$contents = $filesystem->get_contents( $file_path );
+
+		if ( false === $contents ) {
+			return new \WP_Error( 'file_not_read', __( 'The log file could not be read.', 'rocket' ) );
+		}
+
+		$formatter = static::get_stream_formatter();
+		$entries   = 0;
+
+		if ( $formatter ) {
```

Comment:
return early here with a WP_Error if `$formatter` is not ok

Reply:
I can make it simpler by removing completely `get_stream_formatter()`, and just assume that the HTML formatter is used.

Reply:
Even though it's true for now, we might want to offer other formats latter, I think it's good to keep the logic in place